orix/quaternion/quaternion.py

```diff
@@ -498,3 +502,311 @@ def _outer_dask(
+                "passed."
+            )
+        n = euler.shape[:-1]
+        alpha, beta, gamma = euler[..., 0], euler[..., 1], euler[..., 2]
+
+        # Uses A.5 & A.6 from Modelling Simul. Mater. Sci. Eng. 23
+        # (2015) 083501
+        sigma = 0.5 * np.add(alpha, gamma)
+        delta = 0.5 * np.subtract(alpha, gamma)
+        c = np.cos(beta / 2)
+        s = np.sin(beta / 2)
+
+        # Using P = 1 from A.6
+        q = np.zeros(n + (4,))
+        q[..., 0] = c * np.cos(sigma)
+        q[..., 1] = -s * np.cos(delta)
+        q[..., 2] = -s * np.sin(delta)
+        q[..., 3] = -c * np.sin(sigma)
+
+        for i in [1, 2, 3, 0]:  # flip the zero element last
+            q[..., i] = np.where(q[..., 0] < 0, -q[..., i], q[..., i])
+
+        quat = cls(q)
+
+        if direction == "crystal2lab":
+            quat = ~quat
+
+        return quat
+
+    def to_matrix(self) -> np.ndarray:
+        """Return the quaternions as orientation matrices
+        :cite:`rowenhorst2015consistent`.
+
+        Returns
+        -------
+        om
+            Array of orientation matrices.
+
+        Examples
+        --------
+        >>> from orix.quaternion import Quaternion
+        >>> rot = Quaternion([1, 0, 0, 0])
+        >>> np.allclose(rot.to_matrix(), np.eye(3))
+        True
+        >>> rot = Quaternion([0, 1, 0, 0])
+        >>> np.allclose(rot.to_matrix(), np.diag([1, -1, -1]))
+        True
+        """
+        a, b, c, d = self.a, self.b, self.c, self.d
+        om = np.zeros(self.shape + (3, 3))
+
+        bb = b**2
+        cc = c**2
+        dd = d**2
+        qq = a**2 - (bb + cc + dd)
+        bc = b * c
+        ad = a * d
+        bd = b * d
+        ac = a * c
+        cd = c * d
+        ab = a * b
+        om[..., 0, 0] = qq + 2 * bb
+        om[..., 0, 1] = 2 * (bc - ad)
+        om[..., 0, 2] = 2 * (bd + ac)
+        om[..., 1, 0] = 2 * (bc + ad)
+        om[..., 1, 1] = qq + 2 * cc
+        om[..., 1, 2] = 2 * (cd - ab)
+        om[..., 2, 0] = 2 * (bd - ac)
+        om[..., 2, 1] = 2 * (cd + ab)
+        om[..., 2, 2] = qq + 2 * dd
+
+        return om
+
+    @classmethod
+    def from_matrix(cls, matrix: np.ndarray) -> Quaternion:
+        """Return quaternions from the orientation matrices
+        :cite:`rowenhorst2015consistent`.
+
+        Parameters
+        ----------
+        matrix
+            Array of orientation matrices.
+
+        Examples
+        --------
+        >>> from orix.quaternion import Quaternion
+        >>> rot = Quaternion.from_matrix(np.eye(3))
+        >>> np.allclose(rot.data, [1, 0, 0, 0])
+        True
+        >>> rot = Quaternion.from_matrix(np.diag([1, -1, -1]))
+        >>> np.allclose(rot.data, [0, 1, 0, 0])
+        True
+        """
+        om = np.asarray(matrix)
+        # Assuming (3, 3) as last two dims
+        n = (1,) if om.ndim == 2 else om.shape[:-2]
+        quat = np.zeros(n + (4,))
```

Comment:
We can replace this check with

```suggestion
        om = np.atleast_3d(matrix)
        quat = np.zeros(om.shape[:-2] + (4,))
```